routing/road_index.cpp

```diff
@@ -18,6 +31,55 @@ void RoadIndex::Import(vector<Joint> const & joints)
   }
 }
 
+bool RoadIndex::GetAdjacentFtPoint(uint32_t featureIdFrom, uint32_t featureIdTo,
+                                   RestrictionPoint & crossingPoint) const
+{
+  auto const fromIt = m_roads.find(featureIdFrom);
+  if (fromIt == m_roads.cend())
+    return false; // No sense in restrictions to feature which is not road type.
+
+  auto const toIt = m_roads.find(featureIdTo);
+  if (toIt == m_roads.cend())
+    return false; // No sense in restrictions to feature which is not road type.
+
+  RoadJointIds const & roadJointIdsFrom = fromIt->second;
+  RoadJointIds const & roadJointIdsTo = toIt->second;
+  if (roadJointIdsFrom.IsEmpty() || roadJointIdsTo.IsEmpty())
```

Comment:
Please, add a comment why do you check only start and end points only. Also, why not to just create two arrays, each of size 2, the first one will be filled with zero and roadJointIdsFrom.GetSize() - 1, the second one will be filled with zero and roadJointIdsTo.GetSize() - 1 and write a loop over cartesian product of these two arrays. This will prevent you from code duplication.